tools/src/bin/output-file.rs

```diff
@@ -168,8 +168,17 @@ fn read_test_info_from_file_info(per_file_info: Option<&json::Object>) -> Option
     })
 }
 
-fn get_per_file_info<'a>(all_info: &'a json::Object, path: &str) -> Option<&'a json::Object> {
-    let mut cur_obj = all_info.get("root")?.as_object()?;
+/// Per-file info derived from the concise and detailed info for a given file.
+/// Everything in here is optional data, but this structure will be available
+/// for every file to simplify control-flow.
+struct PerFileInfo {
+    bugzilla_component: Option<(String, String)>,
+    test_info: Option<TestInfo>,
+    coverage: Vec<i32>,
```

Comment:
Yeah, semantically `Option<Vec<i32>>` probably makes more sense.  The current structuring was based on me knowing there was no way that I'd create dummy coverage files for every file and that the coverage data uses `-1` to convey that a line isn't covered, so as far as the output logic is concerned, it just treats a lack of data as a `-1`.  But obviously it can do this if there's no array just as well.

I'll plan to make this change, as it likely makes sense to also eventually include more information about the coverage which would live next to the Vec inside the Option.  Like when the coverage run ran, etc.